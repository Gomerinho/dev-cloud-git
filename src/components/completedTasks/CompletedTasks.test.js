import React from 'react';
import { render, screen } from '@testing-library/react';
import CompletedTasks from './index';
import { TasksContext } from '../../utils/contexts/taskContext';

test('renders nothing when there are no completed tasks', () => {
  const tasks = [];
  render(
    <TasksContext.Provider value={{ tasks }}>
      <CompletedTasks />
    </TasksContext.Provider>
  );
  console.log(screen.getByText('Tâches terminées'));
  expect(screen.getByText('Tâches terminées')).toBeInTheDocument();
  expect(screen.getByRole('list')).toBeEmptyDOMElement();
});

test('renders completed tasks correctly', () => {
  const tasks = [
    { id: 1, text: 'Task 1', completed: true },
    { id: 2, text: 'Task 2', completed: true },
  ];

  render(
    <TasksContext.Provider value={{ tasks }}>
      <CompletedTasks />
    </TasksContext.Provider>
  );
  expect(screen.getByText('Task 1')).toBeInTheDocument();
  expect(screen.getByText('Task 2')).toBeInTheDocument();
});

test("doesn't render uncompleted tasks", () => {
  const tasks = [
    { id: 1, text: 'Task 1', completed: true },
    { id: 2, text: 'Task 2', completed: false },
  ];

  render(
    <TasksContext.Provider value={{ tasks }}>
      <CompletedTasks />
    </TasksContext.Provider>
  );
  expect(screen.queryByText('Task 2')).toBeNull();
});

test('matches snapshot', () => {
  const tasks = [
    { id: 1, text: 'Task 1', completed: true },
    { id: 2, text: 'Task 2', completed: true },
  ];

  const { asFragment } = render(
    <TasksContext.Provider value={{ tasks }}>
      <CompletedTasks />
    </TasksContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
