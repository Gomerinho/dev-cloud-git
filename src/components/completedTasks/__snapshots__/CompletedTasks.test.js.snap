// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Tâches terminées
    </h2>
    <ul>
      <li>
        Task 1
      </li>
      <li>
        Task 2
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
